tests/pytest/core/test_middleware_session_req.py

```diff
@@ -0,0 +1,43 @@
+from django.utils.decorators import decorator_from_middleware
+
+from benefits.core import session
+from benefits.core.middleware import AgencySessionRequired
+from benefits.core.models import TransitAgency
+
+import pytest
+
+from tests.pytest.conftest import initialize_request
+
+
+@pytest.mark.django_db
+def test_agency(rf, mocked_view):
+    decorated_view = decorator_from_middleware(AgencySessionRequired)(mocked_view)
+
+    request = rf.get("/something")
+    initialize_request(request)
+
+    agency = TransitAgency.objects.first()
+    session.update(request, agency=agency)
+
+    decorated_view(request)
+    mocked_view.assert_called_once()
```

Comment:
Testing this middleware directly is a bit redundant with the client-level tests below… Thoughts on whether it's useful to keep?

Reply:
Yeah agreed the direct client tests below seem better.

Reply:
So are you planning to remove these `pass` placeholder tests?